### PR TITLE
Support arbitrary path prefix

### PR DESCRIPTION
**Pull request by SMores**
### Description of Change
Library currently expects to be hosted at the root of its domain. However, it can be valuable to host it at a subpath (e.g. mydomain.org/library).

This PR introduces a new environment variable, PATH_PREFIX, which defaults to `'/'`, and formats all internal URLs with that path prefix.

In order to pipe the environment variable through everywhere it needed to go, I had to touch almost every file. If it turns out that there's a cleaner way to do this, please let me know!

This feature is being used live for the NYT Tech Guild's Library site.

This PR also bumps Library up to Node 16; it's currently configured for 10, which is unsupported!

### Checklist

- [x] Ran `npm run lint` and updated code style accordingly
- [x] `npm run test` passes
- [x] PR has a description and all contributors/stakeholder are noted/cc'ed
- [ ] tests are updated and/or added to cover new code
- [ ] relevant documentation is changed and/or added

### CODE REVIEW

**Comment by SMores** [comment]
I could use some help figuring out why there are still a few failing tests on this branch! Some of them seem like they _might_ be related to my code, but some don't!

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.231.6/containers/javascript-node/.devcontainer/base.Dockerfile
```
Comment (afischer) [inline .devcontainer/Dockerfile:1]: Not opposed to leaving this devcontainer stuff in the repo, but if we do, would you mind adding some documentation in the readme about what exactly this is for?

**Comment by afischer** [comment]
Also closes #318.

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 language: node_js
 node_js:
-  - 10
-  - 12
+  - 16
```
Comment (afischer) [inline .travis.yml:3]: Should probably also add 18 now that it's LTS

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM node:12
+FROM node:16
```
Comment (afischer) [inline Dockerfile:1]: ditto on v18 LTS

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (SMores) [reply .devcontainer/Dockerfile:1]: Sure, I can do that! Also happy to drop it if it doesn't seem useful (I just run all of my projects in Docker locally)

File: app.yaml [hunk 4]
```diff
@@ -1,2 +1,2 @@
-runtime: nodejs10
-instance_class: F4
\ No newline at end of file
+runtime: nodejs16
```
Comment (afischer) [inline app.yaml:1]: Here too!

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (afischer) [reply .devcontainer/Dockerfile:1]: I've never used devcontainers and we haven't really experimented with them, so maybe easier to just drop for now if you aren't using?

Comment (SMores) [reply .devcontainer/Dockerfile:1]: I personally am using devcontainers to work on Library, but I'm a regular contributor so, yeah that sounds good!

**Comment by afischer** [comment]
Replaced by #358

**Pull request closed without merging**